frontend/src/components/TreeView.js

```diff
@@ -0,0 +1,179 @@
+                id = FI.find((ticket) => ticket._id === ppr.fi_link).sf_link
+                break
+            case TICKET_TYPES.UPR:
+                let upr = UPR.find((ticket) => ticket._id === ticketID)
+                id = FI.find((ticket) => ticket._id === upr.fi_link).sf_link
+                break
+        }
+        if (id !== -1) setTree(getAllChildren(id, tickets))
+    }, [tickets, location])
+
+    return (
+        <Box
+            pos="fixed"
+            bottom="0"
+            right="0"
+            width="300px"
+            border="1px"
+            borderColor="gray.200"
+            height="auto"
+        >
+            {location.pathname !== '/' && Object.keys(tree).length !== 0 && (
+                <Box padding="24px" width="250px">
+                    <Box bgColor="tomato" cursor="pointer">
+                        <Text
+                            as={
+                                ticketType === 'SF' && ticketID === tree._id
+                                    ? 'b'
+                                    : ''
+                            }
+                            fontSize="xs"
+                            onClick={() => navigate(`/SF/${tree._id}`)}
+                        >
+                            {tree.name}
+                        </Text>
+                    </Box>
+
+                    {sortTickets(tree.fundingItems).map((fi, i) => {
+                        return (
+                            <Box key={i}>
+                                <Box
+                                    pos="relative"
+                                    left="4"
+                                    bgColor="#34cceb"
+                                    cursor="pointer"
+                                >
+                                    <Text
+                                        as={
+                                            ticketType ===
+                                                `${TICKET_TYPES.FI}` &&
+                                            ticketID === fi._id
+                                                ? 'b'
+                                                : ''
+                                        }
+                                        fontSize="xs"
+                                        onClick={() =>
+                                            navigate(
+                                                `/${TICKET_TYPES.FI}/${fi._id}`
+                                            )
+                                        }
+                                    >
+                                        {fi.name}
+                                    </Text>
+                                </Box>
+                                {sortTickets(fi.personalPurchases).map(
+                                    (ppr, j) => {
+                                        return (
+                                            <Box
+                                                key={j}
```

Comment:
Using the key as an index is generally an anti-pattern in react, we should use the code as the key instead (e.g: PPR-2). It can cause issues when the order of elements change, not sure if we encounter that directly with @kcyl9's change, but lets avoid it if we can. Same for the above use of key={i}